### ISSUE TITLE
Missing "Stream size","Bit rate" on/by Opus/vp9 in webm files

### ISSUE THREAD

**Issue by cnleo** [MediaArea/MediaInfo#691]
Missing "Stream size","Bit rate" on/by Opus/vp9 in webm files

```
Format                                   : WebM
Format version                           : Version 4
File size                                : 39.3 MiB
Duration                                 : 4 min 5 s
Overall bit rate                         : 1 343 kb/s
Writing application                      : Lavf59.27.100
Writing library                          : Lavf59.27.100

Video
ID                                       : 1
Format                                   : VP9
Codec ID                                 : V_VP9
Duration                                 : 4 min 5 s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Language                                 : English
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio
ID                                       : 2
Format                                   : Opus
Codec ID                                 : A_OPUS
Duration                                 : 4 min 5 s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Bit depth                                : 32 bits
Compression mode                         : Lossy
Language                                 : English
Default                                  : Yes
Forced                                   : No
```

Maybe relatet bugs:
https://github.com/MediaArea/MediaInfo/issues/271
https://github.com/MediaArea/MediaInfo/issues/348
https://github.com/MediaArea/MediaInfo/issues/540
https://github.com/MediaArea/MediaInfo/issues/558

**Comment by gitoss** [comment]
It's unfortunate that many muxers (incuding ffmpeg) don't add statistic tags to mkv/webm containers, but personally I never had any trouble using "[mkvpropedit](https://mkvtoolnix.download/doc/mkvpropedit.html) --add-track-statistics-tags" and running mediainfo again.

**Comment by jensdraht1999** [comment]
@gitoss I wanted to ask you, does the converter/muxer (for example ffmpeg/handbrake/mkvmuxer) need to save the average bitrate as a statistic in the file or are there other ways to find out the average bitrate of the file by calculation somehow???

**Comment by BlohoJo** [comment]
Bump!

I'm struggling with this as well. It seems that nothing I can do will show me the bitrate of an Opus audio stream inside of an MKV file.

Using latest version 24.04.

**Comment by JeromeMartinez** [comment]
> It seems that nothing I can do will show me the bitrate of an Opus audio stream inside of an MKV file.

Yes, you can, see https://github.com/MediaArea/MediaInfo/issues/691#issuecomment-1503960405.

**Comment by BlohoJo** [comment]
OK, I just figured something out.

The problem seems to be with [yt-dlp](https://github.com/yt-dlp/). I'm using the command line option + argument `--merge-output-format mkv`. The resulting file will only show "Overall bit rate".

If I re-mux using [MKVToolNix](https://mkvtoolnix.download/downloads.html), then it shows the individual bit rates for the audio and video streams.

I created a [feature request](https://github.com/yt-dlp/yt-dlp/issues/9924).

**Comment by jensdraht1999** [comment]
> es the converter/muxer (for example ffmpeg/handbrake/mkvmuxer) need to save the average bitrate as a statistic in the file or are there other ways to find out the average bitrate of the file by calculation somehow???

Just a question, does the muxer or converter has to save it as a statistic to the file? Is this how it has been done always?

**Comment by JeromeMartinez** [comment]
> does the muxer or converter has to save it as a statistic to the file?

The muxer, reason MKVToolNix (which is not a converter) can do it.

> Is this how it has been done always?

Yes.
But not from the origin of Matroska, Matroska itself does not store such metadata, so the developer of MKVToolNix (who is also one of the 2 main Matroska spec authors) created such Matroska tags.
